mne/evoked.py

```diff
@@ -220,11 +251,32 @@ def crop(self, tmin=None, tmax=None, include_tmax=True):
         -----
         %(notes_tmax_included_by_default)s
         """
+        if tmin is None:
+            tmin = self.tmin
+        elif tmin < self.tmin:
+            warn('tmin is not in Evoked time interval. tmin is set to '
+                 'Evoked.tmin')
+            tmin = self.tmin
+
+        if tmax is None:
+            tmax = self.tmax
+        elif tmax > self.tmax:
+            warn('tmax is not in Evoked time interval. tmax is set to '
+                 'Evoked.tmax')
+            tmax = self.tmax
+
         mask = _time_mask(self.times, tmin, tmax, sfreq=self.info['sfreq'],
                           include_tmax=include_tmax)
         self.times = self.times[mask]
         self._update_first_last()
         self.data = self.data[:, mask]
+
+        try:
+            _check_baseline(self.baseline, tmin, tmax, self.info['sfreq'])
+        except ValueError:  # It no longer applies, wipe it out.
```

Comment:
can you try to avoid the try except block? Exception are error prone. It can hide the actual issue.

Reply:
I just copied existing code from the `Epochs` codebase :) But can try to figure out something better.

Reply:
I've tried to improve things in def384d